Automattic/wp-calypso
# Me: Add EditGravatar to profile under feature flag

This PR adds a new component, `EditGravatar`, and uses it in Me > My Profile under a feature flag.

The component lets a user upload a new image as their Gravatar. It will show either a success or failure message.

## How to Test

1. Make sure that nothing is changed on `/me` when the feature flag is turned off
2. Enable the feature flag, and make sure you can see the new section under `/me` using `ENABLE_FEATURES=me/edit-gravatar make run`
3. Get a bearer token using this script:
    ```
    curl -# -X POST --data-urlencode "grant_type=password" --data-urlencode "client_id=appid" --data-urlencode "client_secret=appsecret" --data-urlencode "username=username" --data-urlencode "password=password" https://public-api.wordpress.com/oauth2/token
    ```
    You can use Calypso's app ID & secret, found [here](https://github.com/Automattic/wp-calypso/blob/a1c610c300e040a2c101bb5af6a66f7cd72b52d6/config/production.json#L108-L109).
    If you have 2FA enabled, you can add this extra parameter to the call: `--data-urlencode     "wpcom_otp=one-time-password"`, or you can create a new application password to use instead.
4. Save the bearer token to localStorage using `localStorage.setItem('bearerToken', 'your_bearer_token');`

5. On `/me`, click on 'Select Image' to make sure the component works. You can check if the new Gravatar was uploaded at http://en.gravatar.com/emails/

6. Try to use a bad image, see that you get an error message

7. Watch out for any errors in the console

## Background

This PR is made up of the two first commits from #8456. That PR had to be reverted due to changes made in `<Gravatar />` in https://github.com/Automattic/wp-calypso/pull/8456/commits/4672d9fa6371b1f5590084287eb61c3d71045257. Those changes will be introduced in two new PRs after this one: one for adding the local image to the state, and one for changing `<Gravatar />`.

cc @gwwar
